### PR TITLE
Remove Order#has_available_shipment since it always return nil on val…

### PR DESCRIPTION
**Pull request by tanmay3011**
…idation

### CODE REVIEW

File: core/app/models/spree/order.rb [hunk 1]
```diff
@@ -620,13 +618,6 @@ def ensure_line_items_present
       end
     end
 
-    def has_available_shipment
-      return unless has_step?("delivery")
-      return unless has_step?(:address) && address?
-      return unless ship_address && ship_address.valid?
-      # errors.add(:base, :no_shipping_methods_available) if available_shipping_methods.empty?
```
Comment (nishant-cyro) [inline core/app/models/spree/order.rb:627]: @tanmay3011 Please also remove the message from en.yml.

Comment (nishant-cyro) [inline core/app/models/spree/order.rb:627]: @tanmay3011 - Please add a deprecation warning for this method, rather than removing it.

Comment (tanmay3011) [inline core/app/models/spree/order.rb:627]: @nishant-cyro Then deprecation will be shown in running each spec. Better way of doing it?

**Comment by priyank-gupta** [comment]
@nishant-cyro  I think removing this method should be a better approach than showing warning msg in specs.

**Comment by nishant-cyro** [comment]
@priyank-gupta @tanmay3011 - Lets discuss this tomorrow.

**Comment by nishant-cyro** [comment]
@tanmay3011 @priyank-gupta - we can change deprecation behaviour to :log for test environment, in that case it will log warnings in log file instead of STDERR.

Please refer - http://api.rubyonrails.org/classes/ActiveSupport/Deprecation/Behavior.html

**Comment by nishant-cyro** [comment]
@priyank-gupta @tanmay3011 - There is one more thing we can do is we can remove the method and add this change in Change Log.

Any thoughts?

**Pull request closed without merging**